pkg/storage/azure/azure.go

```diff
@@ -456,35 +465,51 @@ func (d *driver) CreateStorage(cr *imageregistryv1.Config) error {
 			}
 		}
 
-		if d.Config.Container == "" {
+		var containerExists bool
+		if len(d.Config.Container) != 0 {
+			containerExists, err = d.containerExists(d.Config.Container)
+			if err != nil {
+				return err
+			}
+		}
+		if len(d.Config.Container) != 0 && containerExists {
+			cr.Status.Storage.Azure = d.Config.DeepCopy()
+			util.UpdateCondition(cr, imageregistryv1.StorageExists, operatorapiv1.ConditionTrue, storageExistsReasonContainerExists, "Azure container exists")
+		} else {
+			var generatedName bool
+			const numRetries = 5000
 			key, err := d.getAccountPrimaryKey(storageAccountsClient, cfg.ResourceGroup, d.Config.AccountName)
 			if err != nil {
 				util.UpdateCondition(cr, imageregistryv1.StorageExists, operatorapiv1.ConditionFalse, storageExistsReasonAzureError, fmt.Sprintf("Unable to get account primary key: %s", err))
 				return err
 			}
-			const numRetries = 5000
 			for i := 0; i < numRetries; i++ {
 				// If the bucket name is blank, let's generate one
 				if len(d.Config.Container) == 0 {
 					// Container name must be between 3 and 63 characters long
 					if d.Config.Container, err = util.GenerateStorageName(d.Listers, ""); err != nil {
 						return err
 					}
+					generatedName = true
 				}
 
 				err = d.createStorageContainer(d.Config.AccountName, key, d.Config.Container)
 				if err != nil {
 					if e, ok := err.(azblob.StorageError); ok {
 						switch e.ServiceCode() {
 						case azblob.ServiceCodeContainerAlreadyExists:
+							if len(d.Config.Container) != 0 && !generatedName {
+								util.UpdateCondition(cr, imageregistryv1.StorageExists, operatorapiv1.ConditionFalse, "Unable to access container", "The container exists but we do not have permission to access it")
```

Comment:
Convention for condition reason codes is `CamelCase` (and keep it short if we want these to bubble up to telemeter).

```suggestion
								util.UpdateCondition(cr, imageregistryv1.StorageExists, operatorapiv1.ConditionFalse, "StoragePermissionDenied", "The container exists but we do not have permission to access it")
```